qiskit/transpiler/passes/add_barrier_before_measuremets.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+
+"""
+This pass adds a barrier before the measurements.
+"""
+
+from qiskit.extensions.standard.barrier import Barrier
+from qiskit.transpiler import TransformationPass
+from qiskit.dagcircuit import DAGCircuit
+
+
+class AddBarrierBeforeMeasuremets(TransformationPass):
```

Comment:
There's a typo in the class name: missing 'n' in measurements.

Also can we change the name to signify that barriers are only added before final measurements, not all measurements?

Maybe `BarrierBeforeFinalMeasurements`?